src/ngx_http_lua_exten.c

```diff
@@ -0,0 +1,67 @@
+
+/*
+ * Copyright (C) Yichun Zhang (agentzh)
+ * Copyright (C) hongliang5316
+ */
+
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+
+
+#include "ddebug.h"
+#include "ngx_http_lua_exten.h"
+#include "ngx_http_lua_util.h"
+
+
+static int ngx_http_lua_ngx_req_get_uri_exten(lua_State *L);
+
+
+void
+ngx_http_lua_inject_req_exten_api(lua_State *L)
+{
+    lua_pushcfunction(L, ngx_http_lua_ngx_req_get_uri_exten);
+    lua_setfield(L, -2, "get_uri_exten");
+}
+
+
+static int
+ngx_http_lua_ngx_req_get_uri_exten(lua_State *L)
```

Comment:
We no longer add new CFunction-based Lua API to the core.

New API functions should only go the FFI route and be exposed by the lua-resty-core library as a separate Lua module (like the existing `ngx.ssl` and `ngx.semaphore` modules and etc).